src/formatDate.js

```diff
@@ -1,14 +1,34 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      receivedDate['YY'] = dateArr[i].slice(2);
```

Comment:
When you assign 'YY' from 'YYYY', you should also consider that the 'YYYY' key will not be populated in the `receivedDate` object. This will lead to incorrect behavior if 'YYYY' is needed in the `toFormat` array. You should assign the full year to 'YYYY' and then derive 'YY' from it.